kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/job/CDCJob.java

```diff
@@ -200,6 +202,10 @@ public void onFailure(final Throwable throwable) {
             log.error("onFailure, {} task execute failed.", identifier, throwable);
             String jobId = jobItemContext.getJobId();
             jobAPI.updateJobItemErrorMessage(jobId, jobItemContext.getShardingItem(), throwable);
+            if (jobItemContext.getSink() instanceof CDCSocketSink) {
+                CDCSocketSink cdcSink = (CDCSocketSink) jobItemContext.getSink();
+                cdcSink.getChannel().writeAndFlush(CDCResponseUtils.failed("", "", throwable.getMessage()));
```

Comment:
Looks channel.writeAndFlush might be invoked in multi-thread here (CDCSocketSink is still running), could we verify the thread-safety of channel write?

Reply:
I find some related question: 
https://lists.jboss.org/pipermail/netty-users/2008-November/000045.html
https://stackoverflow.com/questions/23161608/is-netty4-channelhandlercontext-write-thread-safe

Reply:
OK